### PR TITLE
Dynamic backup

### PR DESCRIPTION
**Pull request by donomii**
### Review checklist

- [ ] Documentation has been updated, if necessary. Link to changed documentation:
- [ ] Chaos pipeline run or not necessary. Link to pipeline:
- [ ] All new code is covered by tests where it is reasonable.
- [ ] Performance tests have been run or not necessary.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,4 +1,6 @@
 module github.com/weaviate/weaviate
+replace github.com/weaviate/weaviate/entities/models => ./entities/models
+
```
Comment (antas-marcin) [inline go.mod:3]: I think that this needs to be removed

File: openapi-specs/schema.json [hunk 2]
```diff
@@ -4801,6 +4829,24 @@
...
+            "name": "X-Aws-Secret-Key",
+            "in": "header",
+            "required": false,
+            "type": "string"
+          },
+          {
+            "name": "X-Aws-Session-Token",
+            "in": "header",
+            "required": false,
+            "type": "string"
```
Comment (antas-marcin) [inline openapi-specs/schema.json:4849]: I think that we shouldn't add header definitions here in schema, there's lots of other headers that we use beside the AWS ones, so I would not include this information here

File: openapi-specs/schema.json [hunk 3]
```diff
@@ -941,8 +941,20 @@
...
     "BackupConfig": {
       "description": "Backup custom configuration",
-      "type": "object",      
+      "type": "object",
       "properties": {
+        "Endpoint": {
+          "type": "string",
+          "description": "S3 endpoint, e.g. s3.amazonaws.com"
+        },
+        "S3Bucket": {
```
Comment (antas-marcin) [inline openapi-specs/schema.json:950]: hmm, I think that we agreed on different naming, more general: `Bucket`

**Comment by antas-marcin** [comment]
@donomii can we close this PR?

**Comment by donomii** [comment]
obsoleted by https://github.com/weaviate/weaviate/pull/6012

**Pull request closed without merging**